### PR TITLE
adds base cli command for adding routing rules

### PR DESCRIPTION
**Pull request by ilackarms**

### CODE REVIEW

File: cli/pkg/cmd/create/routing_rule.go [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+	}
+	return matchers, nil
+}
+
+func matcherFromInput(input options.RequestMatcher) (*gloov1.Matcher, error) {
+	m := &gloov1.Matcher{}
+	switch {
+	case input.PathExact != "":
+		if input.PathRegex != "" || input.PathPrefix != "" {
+			return nil, errors.Errorf("can only set one of path-regex, path-prefix, or path-exact")
```
Comment (rickducott) [inline cli/pkg/cmd/create/routing_rule.go]: Extract string into constant, write tests for these errors

File: Gopkg.toml [hunk 2]
```diff
@@ -115,3 +115,7 @@
 [[constraint]]
   name = "github.com/solo-io/go-utils"
   version = "v0.2.4"
```
Comment (rickducott) [inline Gopkg.toml]: remove v's in these other ones

File: cli/pkg/options/values.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+	for _, r := range *v {
+		strs = append(strs, fmt.Sprintf("%v.%v", r.Namespace, r.Name))
+	}
+	return "[" + strings.Join(strs, ", ") + "]"
+}
+
+func (v *ResourceRefsValue) Set(s string) error {
+	split := strings.SplitN(s, ".", 2)
+	if len(split) != 2 {
+		return errors.Errorf("%s invalid: refs must be specified in the format <NAMESPACE>.<NAME>", s)
```
Comment (rickducott) [inline cli/pkg/options/values.go]: Add tests for this

File: cli/pkg/options/values.go [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+	for k, val := range *v {
+		strs = append(strs, fmt.Sprintf("%v.%v", k, val))
+	}
+	return "[" + strings.Join(strs, ", ") + "]"
+}
+
+func (v *MapStringStringValue) Set(s string) error {
+	split := strings.SplitN(s, "=", 2)
+	if len(split) != 2 {
+		return errors.Errorf("%s invalid: map entries must be specified in the format KEY=VALUE", s)
```
Comment (rickducott) [inline cli/pkg/options/values.go]: tests

File: cli/pkg/options/values.go [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+	for _, r := range v.Destinations.Destinations {
+		strs = append(strs, fmt.Sprintf("%v: %v", r.Destination.Upstream, r.Weight))
+	}
+	return "[" + strings.Join(strs, ", ") + "]"
+}
+
+func (v *TrafficShiftingValue) Set(s string) error {
+	split := strings.SplitN(s, ".", 2)
+	if len(split) != 2 {
+		return errors.Errorf("%s invalid: weighted destinations must be specified in the format <NAMESPACE>.<NAME>:WEIGHT", s)
```
Comment (rickducott) [inline cli/pkg/options/values.go]: tests

File: cli/pkg/surveyutils/map_string_string_test.go [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/solo-io/gloo/pkg/cliutil/testutil"
+	. "github.com/solo-io/supergloo/cli/pkg/surveyutils"
+)
+
+var _ = Describe("MapStringString", func() {
+	// TODO: unexclude this test when c.ExpectEOF() is fixed
```
Comment (rickducott) [inline cli/pkg/surveyutils/map_string_string_test.go]: link to issue

File: cli/pkg/surveyutils/routing_rule_test.go [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+	. "github.com/onsi/gomega"
+	"github.com/solo-io/gloo/pkg/cliutil/testutil"
+	"github.com/solo-io/solo-kit/pkg/api/v1/clients"
+	"github.com/solo-io/solo-kit/pkg/api/v1/resources/core"
+	"github.com/solo-io/supergloo/cli/pkg/helpers"
+	"github.com/solo-io/supergloo/cli/pkg/options"
+	"github.com/solo-io/supergloo/test/inputs"
+)
+
+// TODO: unexclude this test when c.ExpectEOF() is fixed
```
Comment (rickducott) [inline cli/pkg/surveyutils/routing_rule_test.go]: link issue

File: cli/pkg/cmd/create/routing_rule_test.go [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+			namespace,
+			version string,
+			mtls,
+			autoInject,
+			prometheus,
+			jaeger,
+			grafana bool) {
+
+			err := utils.Supergloo(
+				fmt.Sprintf("create routingrule %v", kind) +
```
Comment (rickducott) [inline cli/pkg/cmd/create/routing_rule_test.go]: what is kind here?

File: cli/pkg/surveyutils/trafficshifting_test.go [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+	"github.com/solo-io/solo-kit/pkg/api/v1/resources/core"
+	"github.com/solo-io/supergloo/cli/pkg/helpers"
+	"github.com/solo-io/supergloo/cli/pkg/options"
+	v1 "github.com/solo-io/supergloo/pkg/api/v1"
+	"github.com/solo-io/supergloo/test/inputs"
+
+	. "github.com/solo-io/supergloo/cli/pkg/surveyutils"
+)
+
+var _ = Describe("Trafficshifting", func() {
```
Comment (rickducott) [inline cli/pkg/surveyutils/trafficshifting_test.go]: does traffic shifting require interactive mode?

Comment (ilackarms) [reply cli/pkg/surveyutils/trafficshifting_test.go]: i'm not sure i understand the question

**Pull request merged**